### PR TITLE
retry: init at 1.04

### PR DESCRIPTION
**Pull request by gfrascadorio**
Add package for 'retry' tool that allows a script to repeat a command
until it succeeds or hits a repetition threshold. 'Retry' supports
replaying stdin, emitting the final stdout, and backoff delays.

###### Motivation for this change

I needed a retry for another package that had to wait on a PostgresQL startup. 
Retry was available in Ubuntu, but not nix.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date (gf - there is a man page and it gets installed too)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123413` at 12090998 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>retry</li>
    <li>tests.trivial</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

automake is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/system/retry/default.nix:6:5:
```
  |
6 |     buildInputs = [ libtool automake autoconf which txt2man ];
  |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

libtool is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/system/retry/default.nix:6:5:
```
  |
6 |     buildInputs = [ libtool automake autoconf which txt2man ];
  |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

which is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/system/retry/default.nix:6:5:
```
  |
6 |     buildInputs = [ libtool automake autoconf which txt2man ];
  |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

autoconf is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/tools/system/retry/default.nix:6:5:
```
  |
6 |     buildInputs = [ libtool automake autoconf which txt2man ];
  |     ^
```
</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3651,6 +3651,12 @@
     githubId = 10353047;
     name = "Tobias Happ";
   };
+  gfrascadorio = {
+    email = "gfrascadorio@tutanota.com";
+    github = "gfrascadorio";
+    githubId = 37602871;
+    name = "Galois";
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:3659]: Separate commit: `maintainers: add <name>`

File: pkgs/tools/system/retry/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchFromGitHub, libtool, automake, autoconf, which, txt2man }:
```
Comment (legendofmiracles) [inline pkgs/tools/system/retry/default.nix:1]: Usually everything is indented by two spaces in nixpkgs, but i don't know if that's a PR blocker.

You can use nixpkgs-fmt to format this for you, it'll save you a lot of time

File: pkgs/tools/system/retry/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+stdenv.mkDerivation rec {
+    pname = "retry";
+    version = "1.04";
+
+    buildInputs = [ libtool automake autoconf which txt2man ];
+
+    src = fetchFromGitHub {
+        owner = "minfrin";
+        repo = "retry";
```
Comment (legendofmiracles) [inline pkgs/tools/system/retry/default.nix:10]:
```suggestion
        repo = pname;
```

File: pkgs/tools/system/retry/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        rev = "f6f1acd4b235e96dd9cb11263fb181ec44c3ece9";
```
Comment (legendofmiracles) [inline pkgs/tools/system/retry/default.nix:11]:
```suggestion
        rev = "v${version}";
```
The commit pulling from should usually be the commit, under which the version was released. 
I haven't checked if they're the same commit here, but the version is always preferred.

**Review by legendofmiracles — CHANGES_REQUESTED** [review]

**Comment by legendofmiracles** [comment]
The commit name/PR name should also follow the [contribution guidelines](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md)

File: pkgs/tools/system/retry/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        sha256 = "sha256:0jrx4yrwlf4fn3309kxraj7zgwk7gq6rz5ibswq3w3b3jfvxi8qb";
+    };
+
+    buildConfigScript = ''
+        libtoolize --force
+        aclocal
+        autoheader
+        automake --add-missing
+        autoconf
+    '';
```
Comment (SuperSandro2000) [inline pkgs/tools/system/retry/default.nix:21]: We can probably replace this with autoreconfHook.

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the styling guide which enforces 2 spaces instead of 4.

**Comment by legendofmiracles** [comment]
> Please follow the styling guide which enforces 2 spaces instead of 4.
I already said this, but I'll just say it again - you can use `nixpkgs-fmt` to format the code for you, it'll save you a lot of time

File: pkgs/tools/system/retry/default.nix (hunk 5 above)
Comment (gfrascadorio) [reply pkgs/tools/system/retry/default.nix:21]: Replaced.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (gfrascadorio) [reply maintainers/maintainer-list.nix:3659]: Separate commit or separate PR too?

Comment (legendofmiracles) [reply maintainers/maintainer-list.nix:3659]: Just a commit.

**Comment by legendofmiracles** [comment]
Squash all related commits into one, when you're done

**Comment by minfrin** [comment]
Not sure if you monitor for new releases. v1.0.5 is available:

https://github.com/minfrin/retry/releases/tag/retry-1.0.5

**Comment by SuperSandro2000** [comment]
the updater bot should pick that up soon

**Pull request merged**